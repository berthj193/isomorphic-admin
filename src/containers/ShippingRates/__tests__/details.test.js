import React from 'react';
import ShippingRateDetails, { ShippingRateDetails as RawShippingRateDetails } from '../details';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';

describe('<ShippingRateDetails />', () => {
  const defaultParams = {
    match: {
      params: {},
    },
    shipping_rates: [{}],
  };
  const getComponent = (props = {}) => shallow(<RawShippingRateDetails {...defaultParams} {...props} />);
  const mockStore = configureStore();

  it('should render properly', () => {
    expect(toJson(getComponent({ shipping_rates: [{ id: '1' }] }))).toMatchSnapshot();
  });

  it('should render loading page', () => {
    expect(toJson(getComponent())).toMatchSnapshot();
  });

  it('should render details with provider', () => {
    expect(toJson(shallow(<ShippingRateDetails store={mockStore()} />))).toMatchSnapshot();
  });
});
