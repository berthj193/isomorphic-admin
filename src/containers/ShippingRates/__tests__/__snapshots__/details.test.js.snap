// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShippingRateDetails /> should render details with provider 1`] = `
<ComponentWithProvider
  startRequest={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;

exports[`<ShippingRateDetails /> should render loading page 1`] = `
<React.Fragment>
  <LayoutWrapper
    style={
      Object {
        "minHeight": "100vh",
      }
    }
  >
    <LayoutContent>
      <h1>
        Warehouses
      </h1>
      <ButtonList
        buttons={
          Array [
            Object {
              "link": "/dashboard/shipping_rates",
              "title": "Go back",
            },
            Object {
              "link": "/dashboard/shipping_rates/undefined/edit",
              "title": "Edit",
            },
          ]
        }
      />
      <PageDetails
        details={
          Array [
            Object {
              "Country": undefined,
            },
            Object {
              "Name": undefined,
            },
            "---",
            Object {
              "Is premium": "No",
            },
            Object {
              "Is live": "No",
            },
            "---",
            Object {
              "Updated at": undefined,
            },
          ]
        }
      />
    </LayoutContent>
  </LayoutWrapper>
</React.Fragment>
`;

exports[`<ShippingRateDetails /> should render properly 1`] = `
<PageLoading
  style={
    Object {
      "height": "50vh",
    }
  }
/>
`;
